shell.py

```diff
@@ -311,6 +311,33 @@ def help_ls(self):
         self.stdout.write('List directory contents.\n' +
                           'Use ls -a to show hidden files\n')
 
+def help_cp(self):
+    self.stdout.write('Copies contents of one text file to a new text file\n' +
+                        'syntax: cp source.py destination.py\n')
+
+def do_cp(self, line):
+    args = self.line_to_args(line)
+    source = self.resolve_path(args[0])
+    mode = get_mode(source)
+    if not mode_exists(mode):
+        self.stderr.write("Cannot access '%s': No such file\n" % source)
+        return 0
+    if not mode_isfile(mode):
+        self.stderr.write("'%s': is not a file\n" % filename)
+        return 0
+    destination = self.resolve_path(args[1])
```

Comment:
The code should probably verify that exactly 2 arguments were passed. The regular cp command allows:
```
cp file1 file2 file3 directory
```
and if somebody entered that, we should generate an error rather than causing file1 to overwrite file2.